AutomationHelpers/src/Modules/CollectRetryErrorMessagesModule.cs

```diff
@@ -0,0 +1,143 @@
+﻿//
+// Copyright © 2017 Ranorex All rights reserved
+//
+
+using System;
+using System.Collections.Generic;
+using Ranorex.Core.Testing;
+using Ranorex.Core.Reporting;
+
+namespace Ranorex.AutomationHelpers.Modules
+{
+    /// <summary>
+    /// Functionality: Collects all error messages from given test container.

```

Comment:
When placed in a test container with a retry count >= 1, collects all error messages even if one of the retries was successful and adds them to the report.